comp/metadata/packagesigning/packagesigningimpl/packagesigning.go

```diff
@@ -128,11 +130,29 @@ func newPackageSigningProvider(deps dependencies) provides {
 }
 
 func isPackageSigningEnabled(conf config.Reader, logger log.Component) bool {
-	if !conf.GetBool("enable_signing_metadata_collection") {
-		logger.Debug("Signing metadata collection disabled: linux package signing keys will not be collected nor sent")
+	installTool, err := installinfo.Get(conf)
+	if err != nil {
+		logger.Debugf("Failed to get install_info file information: %v", err)
 		return false
 	}
+	isInConfigurationFile := conf.GetBool("enable_signing_metadata_collection")
+	isEnabled := isInConfigurationFile && runtime.GOOS == "linux" && isAllowedInstallationTool(installTool.Tool)
+	if !isEnabled {
+		logger.Debugf("Package signing metadata collection disabled: config %t, OS %s, install tool %s", isInConfigurationFile, runtime.GOOS, installTool.Tool)
+		logger.Debug("Package signing telemetry must be enabled in datadog.yaml, and running on a non-containerized Linux system to collect data")
+	}
 	logger.Debug("Signing metadata collection enabled")
```

Comment:
nit: shoudln't this be in an `else` block?

Reply:
correct, thanks